logger/logfile.go

```diff
@@ -76,11 +87,39 @@ func (l *LogFile) rotate() error {
 	// Rotate if we hit the byte file limit or the time limit
 	if (l.BytesWritten >= int64(l.MaxBytes) && (l.MaxBytes > 0)) || timeElapsed >= l.duration {
 		l.FileInfo.Close()
+		if err := l.purgeArchivesIfNeeded(); err != nil {
+			fmt.Printf("E! Unable to purge old log file archives: %s", err.Error())
+		}
 		return l.openNew()
 	}
 	return nil
 }
 
+func (l *LogFile) purgeArchivesIfNeeded() error {
+	if l.MaxLogArchives == -1 {
+		return nil
+	}
+	fileNamePattern := l.getFileNamePattern()
+	//get all the files that match the log file pattern
+	globExpression := filepath.Join(l.logPath, fmt.Sprintf(fileNamePattern, "*"))
+	var matches []string
+	var err error
+	if matches, err = filepath.Glob(globExpression); err != nil {
+		return err
+	}
+	//if there are more archives than the configured maximum, then purge
+	if len(matches) > l.MaxLogArchives {
```

Comment:
```golang
stale := len(matches) - l.MaxFiles
for i := 0; i < stale; i++ {
    if err := os.Remove(matches[i]); err != nil {
        return err
    }
}
```

This is all we need to do since `Glob` is returning the results from one directory sorted already: https://golang.org/src/path/filepath/match.go?s=5600:5655#L325 and loop makes the conditional obsolete.